langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiCompletionModel.java

```diff
@@ -0,0 +1,171 @@
+     * @param randomSeed    the random seed for generating responses
+     * @param stops         a list of tokens at which the model should stop generating tokens
+     * @param timeout       the timeout duration for API requests
+     *                      <p>
+     *                      The default value is 60 seconds.
+     * @param logRequests   a flag indicating whether to log API requests
+     * @param logResponses  a flag indicating whether to log API responses
+     * @param maxRetries    the maximum number of retries for API requests. It uses the default value 3 if not specified.
+     */
+    @Builder
+    public MistralAiCompletionModel(String baseUrl,
+                                    String apiKey,
+                                    String modelName,
+                                    Double temperature,
+                                    Integer maxTokens,
+                                    Integer minTokens,
+                                    Double topP,
+                                    Integer randomSeed,
+                                    List<String> stops,
+                                    Duration timeout,
+                                    Boolean logRequests,
+                                    Boolean logResponses,
+                                    Integer maxRetries) {
+
+        this.client = MistralAiClient.builder()
+                .baseUrl(getOrDefault(baseUrl, "https://api.mistral.ai/v1"))
+                .apiKey(apiKey)
+                .timeout(getOrDefault(timeout, Duration.ofSeconds(60)))
+                .logRequests(getOrDefault(logRequests, false))
+                .logResponses(getOrDefault(logResponses, false))
+                .build();
+        this.modelName = getOrDefault(modelName, MistralAiCodeModelName.CODESTRAL_LATEST.toString());
+        this.suffix = "";
+        this.temperature = temperature;
+        this.maxTokens = maxTokens;
+        this.minTokens = getOrDefault(minTokens,0);
+        this.topP = topP;
+        this.randomSeed = randomSeed;
+        this.stops = stops;
+        this.maxRetries = getOrDefault(maxRetries, 3);
+    }
+
+    /**
+     * Creates a MistralAiCompletionModel with the specified API key.
+     *
+     * @param apiKey the API key for authentication
+     * @return a MistralAiCompletionModel
+     */
+    public static MistralAiCompletionModel withApiKey(String apiKey) {
+        return builder().apiKey(apiKey).build();
+    }
+
+    /**
+     * Generates a completion for the given prompt and suffix.
+     *
+     * @param prompt the starting point of the text/code
+     * @param suffix Optional text/code that adds more context for the model. When given a prompt and a suffix the model will fill what is between them.
+     * @return a response containing the generated text/code
+     */
+    public Response<String> generate(String prompt, String suffix) {
+        this.suffix = suffix;
```

Comment:
this does not seem right. If `generate(String)` is called afterwards, it will use the same suffix which is probably not how it sohuld work